cli/js/format_error_test.ts

```diff
@@ -0,0 +1,236 @@
+    lineNumber: 1000,
+    scriptResourceName: "foo.ts",
+    startColumn: 1,
+    endColumn: 2,
+    code: 4000
+  }
+];
+
+const expected01 = `[1;31merror[0m[1m TS4000[0m: Example error
+
+► [38;5;14mfoo.ts[0m:[38;5;11m1001[0m:[38;5;11m2[0m
+
+[47;30m1001[0m abcdefghijklmnopqrstuv
+[47;30m    [0m [31m ^[0m
+`;
+
+const fixture02: Deno.DiagnosticItem[] = [
+  {
+    message: "Example error",
+    category: Deno.DiagnosticCategory.Error,
+    sourceLine: "abcdefghijklmnopqrstuv",
+    lineNumber: 1000,
+    scriptResourceName: "foo.ts",
+    startColumn: 1,
+    endColumn: 5,
+    code: 4000
+  }
+];
+
+const expected02 = `[1;31merror[0m[1m TS4000[0m: Example error
+
+► [38;5;14mfoo.ts[0m:[38;5;11m1001[0m:[38;5;11m2[0m
+
+[47;30m1001[0m abcdefghijklmnopqrstuv
+[47;30m    [0m [31m ~~~~[0m
+`;
+
+const fixture03: Deno.DiagnosticItem[] = [
+  {
+    message: "Example error",
+    messageChain: {
+      message: "First level",
+      category: Deno.DiagnosticCategory.Error,
+      code: 4001,
+      next: [
+        {
+          message: "Level 2 01",
+          category: Deno.DiagnosticCategory.Error,
+          code: 4002
+        },
+        {
+          message: "Level 2 02",
+          category: Deno.DiagnosticCategory.Error,
+          code: 4003,
+          next: [
+            {
+              message: "Level 3 01",
+              category: Deno.DiagnosticCategory.Error,
+              code: 4004
+            }
+          ]
+        }
+      ]
+    },
+    category: Deno.DiagnosticCategory.Error,
+    sourceLine: "abcdefghijklmnopqrstuv",
+    lineNumber: 1000,
+    scriptResourceName: "foo.ts",
+    startColumn: 1,
+    endColumn: 5,
+    code: 4000
+  }
+];
+
+const expected03 = `[1;31merror[0m[1m TS4000[0m: First level
+  Level 2 01
+  Level 2 02
+    Level 3 01
+
+► [38;5;14mfoo.ts[0m:[38;5;11m1001[0m:[38;5;11m2[0m
+
+[47;30m1001[0m abcdefghijklmnopqrstuv
+[47;30m    [0m [31m ~~~~[0m
+`;
+
+const fixture04: Deno.DiagnosticItem[] = [
+  {
+    message: "Example error",
+    relatedInformation: [
+      {
+        message: "Related information",
+        category: Deno.DiagnosticCategory.Info,
+        sourceLine: "1234567890",
+        lineNumber: 99,
+        scriptResourceName: "foo.ts",
+        startColumn: 3,
+        endColumn: 7,
+        code: 1000
+      }
+    ],
+    category: Deno.DiagnosticCategory.Error,
+    sourceLine: "abcdefghijklmnopqrstuv",
+    lineNumber: 1000,
+    scriptResourceName: "foo.ts",
+    startColumn: 1,
+    endColumn: 5,
+    code: 4000
+  }
+];
+
+const expected04 = `[1;31merror[0m[1m TS4000[0m: Example error
+
+► [38;5;14mfoo.ts[0m:[38;5;11m1001[0m:[38;5;11m2[0m
+
+[47;30m1001[0m abcdefghijklmnopqrstuv
+[47;30m    [0m [31m ~~~~[0m
+
+  Related information
+
+    ► [38;5;14mfoo.ts[0m:[38;5;11m100[0m:[38;5;11m4[0m
+
+    [47;30m100[0m 1234567890
+    [47;30m   [0m [38;5;14m   ~~~~[0m
+
+`;
```

Comment:
It kinda sucks that these outputs are hard to read with the colors. It also seems like this test will be quite brittle when the formatting changes. Maybe instead of checking the exact output you can use String.includes to check for various expected substrings?

Reply:
I tried stripping the ANSI codes using a regex, it didn't work.  I should try again.  Doing just `.includes()` feels like it would not be good coverage.

Reply:
I think it is. If we're getting a string and it's not empty and kinda matches what we think, then it will almost match the formatting on the rust side - where we can have more detailed tests.